data/accountManager.go

```diff
@@ -61,6 +62,19 @@ func MakeAccountManager(log logging.Logger, registry account.ParticipationRegist
 	return manager
 }
 
+// DeleteStateProofKeysForExpiredAccounts removes ephemeral keys for every expired account
+func (manager *AccountManager) DeleteStateProofKeysForExpiredAccounts(currentRound basics.Round) {
+	for _, part := range manager.registry.GetAll() {
```

Comment:
Here, `GetAll` calls `record.Duplicate()` on each record, and most of them will be discarded after failing `currentRound <= part.LastValid` check. 

Is it worth passing a parameter `currentRound` down to `GetAll` to do the check and only return the results which are relevant here?

Reply:
It looks like calling `GetAll` is a common pattern when interacting with the registry. Apparently, the idea was to create simple interfaces. We can create an alternative interface, but since this code is being invoked every ~256 rounds, it might not be worth introducing more complex functions.

Reply:
1. It should not be a complex function. The check in the outer function loop will be pushed to the internal function look. 
2. Under normal circumstances, it will be irrelevant, since it will only be every ~256 rounds. However, there is a plan to make this period shorter, and, more importantly, this could be unnecessary cost during catchup, where these calls will be very frequent.

Reply:
https://github.com/algorand/go-algorand/blob/4bbedc00180fb4e20a3496ba5ee8a31011d29159/data/account/participationRegistry.go#L593
This is happening every round as well. I don't see a reason why this code should be different.
If you think we should change it, let's open a ticket and change both of them.

Reply:
going to change this code a bit. let's move the discussion to the new PR.